src/Network/Transport/TCP.hs

```diff
@@ -1717,7 +1743,9 @@ runScheduledAction (ourEndPoint, theirEndPoint) mvar = do
     handleIOException ex vst = do
       -- Release probing resources if probing.
       forM_ (remoteProbing vst) id
-      tryCloseSocket (remoteSocket vst)
```

Comment:
A similar question arises here. If an error occurs, the thread running `handleIncomingMessages` used to terminate when reading from the closed socket. Now it is left running.